source/lac/petsc_precondition.cc

```diff
@@ -488,32 +490,62 @@ namespace PETScWrappers
     AssertThrow (ierr == 0, ExcPETScError(ierr));
 
     if (additional_data.output_details)
-      PetscOptionsSetValue("-pc_hypre_boomeramg_print_statistics","1");
+#if PETSC_VERSION_GT(3,7,0)
+      PetscOptionsSetValue(NULL, "-pc_hypre_boomeramg_print_statistics","1");
 
-    PetscOptionsSetValue("-pc_hypre_boomeramg_agg_nl",
+    PetscOptionsSetValue(NULL, "-pc_hypre_boomeramg_agg_nl",
                          Utilities::int_to_string(
                            additional_data.aggressive_coarsening_num_levels
                          ).c_str());
+#else
+      PetscOptionsSetValue("-pc_hypre_boomeramg_print_statistics","1");
+
+    PetscOptionsSetValue("-pc_hypre_boomeramg_agg_nl",
+			 Utilities::int_to_string(
+                         additional_data.aggressive_coarsening_num_levels
+                         ).c_str());
+
+#endif
```

Comment:
This piece of code is very difficult to read, by the way because you have
```
  if (cond)
#if ...
    statement;

  another statement
#else
    statement;

 another statement;
```
In other word, the `#if` intercepts an `if(cond)` statement and continuing code. Can you just duplicate the `if (additional_data.output_details)` condition in both the `#if` and `#else` branches to make it easier to read this?